log/log.go

```diff
@@ -142,9 +143,23 @@ func Debugf(msg string, args ...interface{}) {
 	logger.Debugf(msg, args...)
 }
 
-type nilLogger struct {
+var levelPriorities = map[Level]int{
+	DebugLevel: 0,
```

Comment:
We have declared the levels above and also we introduced a global map of levels assigned to integers. Is it possible to merge these two and avoid multiple definition of log levels? Also if this is only used inside this package's scope let's avoid the global variable

Reply:
The first declaration of levels is about having exported constants while the mapping to integers is only for internal use. I think if we try to merge those we are going to decrease readability and clarity. Note also that there are direct usages of level constants in our code, I think we don't want to break these.
About the global variable, as long as it is unexported I think we are pretty much OK. What do you think?